pkg/networkmanager/firewall.jsx

```diff
@@ -589,6 +611,113 @@ class RemoveServicesModal extends React.Component {
     }
 }
 
+class ActivateZoneModal extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            ipRange: "",
+            ipRangeValue: "",
+            zone: "",
+            interfaces: [],
+        };
+        this.onInterfaceChange = this.onInterfaceChange.bind(this);
+        this.onChange = this.onChange.bind(this);
+        this.save = this.save.bind(this);
+    }
+
+    onInterfaceChange(event) {
+        let int = event.target.value;
+        this.setState(state => {
+            let interfaces = int === "all" ? [] : state.interfaces;
+            if (state.interfaces.indexOf(int) === -1)
+                return { interfaces: interfaces.concat(int) };
+            return { interfaces: interfaces.filter(i => i !== int) };
+        });
+    }
+
+    onChange(key, value) {
+        this.setState({ [key]: value });
+    }
+
+    save() {
+        let interfaces = this.state.interfaces.concat().filter(i => i !== "all");
+        let sources = this.state.ipRange === "ip-range" ? this.state.ipRangeValue.split(",").map(ip => ip.trim()) : [];
+        firewall.activateZone(this.state.zone, interfaces, sources).then(() => this.props.close());
+    }
+
+    render() {
+        let zones = Object.keys(firewall.zones).filter(z => firewall.zones[z].target === "default" && !firewall.activeZones.has(z));
+        return (
+            <Modal id="add-zone-dialog" show onHide={this.props.close}>
+                <Modal.Header>
+                    <Modal.Title>{ _("Add Zone") }</Modal.Title>
+                </Modal.Header>
+                <Modal.Body id="add-zone-dialog-body">
+                    <form className="ct-form-layout">
+                        <label htmlFor="add-zone-services-readonly" className="control-label">
+                            <div className="vertical">{ _("Trust level") }<small>{ _("Sorted from least") }<br />{ _("trust to most trust") }</small></div>
```

Comment:
This kind of static text breaking doesn't work with i18n. This needs to rely on CSS automatically breaking the words, and using `max-width:`.